drivers/sensor/nordic/qdec_nrfx/qdec_nrfx.c

```diff
@@ -251,13 +252,15 @@ static int qdec_nrfx_pm_action(const struct device *dev, enum pm_device_action a
 static int qdec_nrfx_init(const struct device *dev)
 {
 	const struct qdec_nrfx_config *config = dev->config;
-	nrfx_err_t nerr;
+	struct qdec_nrfx_data *dev_data = dev->data;
+	int nerr;
 
 	config->irq_connect();
 
-	nerr = nrfx_qdec_init(&config->qdec, &config->config, qdec_nrfx_event_handler, (void *)dev);
-	if (nerr != NRFX_SUCCESS) {
-		return (nerr == NRFX_ERROR_INVALID_STATE) ? -EBUSY : -EFAULT;
+	nerr = nrfx_qdec_init(&dev_data->qdec, &config->config, qdec_nrfx_event_handler,
+		(void *)dev);
+	if (nerr != 0) {
+		return -EALREADY;
```

Comment:
Previously this was `EBUSY` or `EFAULT`, why change that?

Reply:
I've made it to simplify error handling a bit, because there were two completely different error spaces